### PR TITLE
Create Vernon README and simple Vernon experiment example

### PR DESCRIPTION
**Pull request by nielsleadholm**
projects/vernon_examples was created to enable adding other, more complex experiment examples in the future (as advised by Subutai)

### CODE REVIEW

File: nupic/research/frameworks/vernon/README.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
+Vernon
+==============
+Vernon is a flexible framework to allow researchers to explore neural network models under a variety of learning paradigms including continuous, meta, and sparse learning. It is designed to be light-weight with a focus on mixin classes.
```
Comment (lucasosouza) [inline nupic/research/frameworks/vernon/README.md:3]: Sparse learning here, compared to meta and continuous learning, implies it is a training paradigm. The term "sparse learning" would be more adequate for something like learning with sparse rewards or sparse observations.

What we aim do is to support different learning paradigms (unsupervised, supervised, continual, meta, etc.) using networks which are biologically inspired, including for example sparse neural networks.

File: nupic/research/frameworks/vernon/README.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+Vernon is a flexible framework to allow researchers to explore neural network models under a variety of learning paradigms including continuous, meta, and sparse learning. It is designed to be light-weight with a focus on mixin classes. 
+
+Vernon is the creation of researchers at Numenta; note therefore that all of the caveats specified in the nupic.research README still apply. 
+
+Getting started
+==============
+To get started, the key modules to be aware of are
+- [handlers.py](./handlers.py) : contains classes to structure the desired experiment (such as supervised vs continual learning)
+- [run.py](./run_experiment/run.py) and [run_with_raytune.py](./run_experiment/run_with_raytune.py) : contain functions to run more advanced experiments
+- [common_models.py](../pytorch/models/common_models.py) in our PyTorch framework : contains some common models that you can use
```
Comment (lucasosouza) [inline nupic/research/frameworks/vernon/README.md:12]: Change the third person to passive voice to make it more formal, from "contains some common models that you can use " to "contain common models that can be used to..."

File: nupic/research/frameworks/vernon/README.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+To get started, the key modules to be aware of are
+- [handlers.py](./handlers.py) : contains classes to structure the desired experiment (such as supervised vs continual learning)
+- [run.py](./run_experiment/run.py) and [run_with_raytune.py](./run_experiment/run_with_raytune.py) : contain functions to run more advanced experiments
+- [common_models.py](../pytorch/models/common_models.py) in our PyTorch framework : contains some common models that you can use 
+
+Running a basic experiment
+==============
+On your local machine you can run simple_experiment.py found under nupic.research/projects/vernon_examples/
+```bash
+python ./simple_experiment.py
```
Comment (lucasosouza) [inline nupic/research/frameworks/vernon/README.md:18]: No need for the `./`

File: nupic/research/frameworks/vernon/README.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+	- for epoch 1 to e:
+		- for batch 1 to b:
+			- gradient_step()
+	- evaluate_task()
+	- evaluate_all_previous_tasks()
+
+
+Mixins for advanced experiments
+==============
+Mixins enable one to specificlaly inherit desired features from a particular parent class, without creating a rigid relationship between the parent and child. In Vernon this enables one to flexibly define new experiments. Depending on the execution order of the defined experiment, this will determine the class properties that are inherited and therefore the experiment that will be performed.
```
Comment (lucasosouza) [inline nupic/research/frameworks/vernon/README.md:47]: Typo: "specifically"

File: nupic/research/frameworks/vernon/README.md [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+
+Vernon provides a variety of mixins that enable the specification of more advanced experiments, including:
+- custom loss functions (composite_loss.py)
+- knowledge distillation to have the network learn from a teacher model (knowledge_distillation.py)
+- efficient tuning of the learning rate (lr_range_test.py)
+- MaxUp data augmentation to improve model generalization (maxup.py)
+
+About the name
+==============
+Vernon is a reference to [Vernon Mountcastle](https://en.wikipedia.org/wiki/Vernon_Benjamin_Mountcastle), affectionately known as 'Mount Vernon' around Numenta, who first discovered and characterized the columnar organization of the neo-cortex.
```
Comment (lucasosouza) [inline nupic/research/frameworks/vernon/README.md:86]: Hahahaha I didn't know about the Mount Vernon reference

File: projects/vernon_examples/simple_example.py [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+    # neural networks in supervised learning tasks
+    optimizer_class=torch.optim.Adam,
+    optimizer_args=dict(
+        lr=1e-4
+    ),  # Optimizer_args is used to initialize the optimizer, so any param that e.g.
+    # Adam takes as an argument can be included
+)
+
+
+def run_experiment(config):
```
Comment (lucasosouza) [inline projects/vernon_examples/simple_example.py:58]: Maybe remove some of the prints and code lines to clean up this function. The lines that track time and print duration for example are not necessary in an example file. Same for the lines that print starting epoch and finished epoch, it is enough to have the prints that start and finish training

**Review by lucasosouza — CHANGES_REQUESTED** [review]

File: projects/vernon_examples/simple_example.py (hunk 6 above)
Comment (nielsleadholm) [reply projects/vernon_examples/simple_example.py:58]: Good point, would you be happy with the following output? As a new user I feel it's helpful to see that the training is progressing before it's finished. It just prevents the user from wondering if it's hung up on something. 

Training started....
Finished Epoch: 1
Accuracy: 0.9129
Finished Epoch: 2
Accuracy: 0.9260
Finished Epoch: 3
Accuracy: 0.9348
....Training finished

**Review by lucasosouza — APPROVED** [review]
